EP-5546: Widget Insertion Refresh Part 2

Mwindo opened the pull request:
Somehow the `pageshow` event is being triggered too early in some cases; setting a higher timeout fixed the issue, but that's obviously a hack.

I feel doubtful about finding the "right" event to guarantee that WordPress has fully "loaded" if `pageshow` fails; instead, it feels safer to reference WordPress functionality directly. This alternative solution is based on https://gist.github.com/KevinBatdorf/fca19e1f3b749b5c57db8158f4850eff. (https://github.com/WordPress/gutenberg/issues/8379 was also helpful.)

https://github.com/orgnc/wordpress-plugin/assets/47676832/a43af454-3578-426a-96ff-b11d4e3abaee

Mwindo commented on src/blocks/affiliate/shared/helpers.js:
```diff
@@ -1,5 +1,36 @@
+
+export const refreshAffiliateWidgetsOnSave = () => {
+  whenEditorIsReady().then(() => {
+    if (!window.wp_organic_affiliate_processed) {
```
I can add a comment, something like, "Prevent multiple calls to pageProcess."

Mwindo replied:
(Since every widget on the admin page will call this.)

sergunich replied:
Adding a comment sounds like a good idea; another suggestion is to use `window.__wpOrganicAffiliateProcessed` flag for consistency

sergunich reviewed (approved):
👍

Mwindo commented:
> The fix for refreshing the widgets on page load works in Chrome/Edge, but it doesn't fix the issue in Safari/Firefox :/ So more to come.

False alarm. After a series of tests, I've confirmed this had to do with page caching. That said, I've refactored out the affiliate toolbar (should have been like this anyway) and added a "Refresh display" option as a last resort.

Mwindo commented:
Toolbar looks like this now:
<img width="1157" alt="Screen Shot 2023-05-22 at 3 11 56 PM" src="https://github.com/orgnc/wordpress-plugin/assets/47676832/599aaa2d-83fa-407d-bcda-0aaba2b42931">
